### PR TITLE
- changes to be able to override methods in blacoilmodel

### PR DESCRIPTION
**Pull request by hnil**
- chaches to update full intensive quantities including extxrution factor without elementctx

### CODE REVIEW

**Comment by bska** [comment]
jenkins build this please

File: opm/models/discretization/common/fvbaseintensivequantities.hh [hunk 1]
```diff
@@ -80,14 +80,15 @@ public:
      */
     void checkDefined() const
     { }
-
+protected:
+    Scalar extrusionFactor_;
```
Comment (atgeirr) [inline opm/models/discretization/common/fvbaseintensivequantities.hh:84]: Why did you need to make this protected? The rest of the changes I believe are good. (But @akva2 should confirm.)

Comment (akva2) [reply opm/models/discretization/common/fvbaseintensivequantities.hh:84]: as this is ebos serialization and not "my" stuff, i don't have much to say as such. but looks like the standard upcasting stuff so it is probably fine.

Comment (bska) [reply opm/models/discretization/common/fvbaseintensivequantities.hh:84]: > Why did you need to make this protected?

The [description](https://github.com/OPM/opm-models/pull/791#issue-1606296298) says that this is needed to update the full set of intensive quantities, **including the extrusion factor**, in a situation that does not have an element context.  For what it's worth, I'm very much in favour of bringing this change in.

**Review by bska — APPROVED** [review]
I think this looks good and certainly generalises the current implementation.  I'll merge this unless there's a **really** good reason to avoid the extrusion factor being promoted from `private` to `protected` status.

**Comment by hnil** [comment]
the extrusionFactor_ think is since to be able ti initializ this in a subclass. This is usefull to be able to have a valid object even if one do not use elemtctx

**Comment by atgeirr** [comment]
Initializing from a subclass is a good reason, thanks. Merging!
